roles/varnish/templates/etc/varnish/varnish.vcl

```diff
@@ -108,7 +108,14 @@ acl purge {
     "localhost";
     "127.0.0.1";
     "{{ frontend_domain }}";
-    # TODO Include IP addresses of other serving hosts.
+{% set frontend_ips = groups.frontend|map('extract', hostvars, ['ansible_default_ipv4', 'address'])|list %}
+{% for host in groups.all %}
+{% if hostvars[host].ansible_default_ipv4.address not in frontend_ips %}
+{# we have to exclude the frontend ip address because otherwise
+   "curl -X PURGE https://cnx.org/" would work from any hosts #}
+    "{{ hostvars[host].ansible_default_ipv4.address }}";
```

Comment:
Ah, because of the 443 -> 80 rewrite. We could fix that in the test by looking if there is an `x-forwarded-for` header, I suppose … But this works, at the risk of making purging hard on single-machine-deploys, no?

Reply:
It should work as it'll match `localhost`.

Reply:
Not sure this is true - doesn't your local setup have `frontend_domain` aliased to `localhost` on /etc/hosts? Just tested locally myself:
```reedstrm@sensei:~/src/rewrite/cnx-publishing (direct-results-access $%)$ python -m SimpleHTTPServer
Serving HTTP on 0.0.0.0 port 8000 ...
10.64.71.229 - - [29/Nov/2017 11:11:22] "GET / HTTP/1.1" 200 -
10.64.71.229 - - [29/Nov/2017 11:11:23] code 404, message File not found
10.64.71.229 - - [29/Nov/2017 11:11:23] "GET /favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [29/Nov/2017 11:11:37] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [29/Nov/2017 11:11:37] code 404, message File not found
127.0.0.1 - - [29/Nov/2017 11:11:37] "GET /favicon.ico HTTP/1.1" 404 -
```
That's two successive hits on `http://10.64.71.229:8000` and `http://localhost:8000`
Again not a huge deal, it means to purge from the frontend, you have to use localhost